### ISSUE TITLE
AdamW algorithm is not the same as in the referenced paper

### ISSUE THREAD

**Issue by quantitative-technologies** [huggingface/transformers#18757]
### System Info

Not relevant

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

The HuggingFace implementation of AdamW is not the same as the algorithm from the paper [Decoupled Weight Decay Regularization](https://arxiv.org/abs/1711.05101), as claimed in the documentation. 

It is easy to show this by mathematical proof, it it is not obvious by inspection. The HuggingFace AdamW has been deprecated in any case. 

But this is still a bug which "caught" me. The issue is referenced, but not as a bug, in the closed issue  [#3407](https://github.com/huggingface/transformers/issues/3407).

### Expected behavior

The PyTorch implementation matches the algorithm in the paper.

**Comment by LysandreJik** [comment]
Hey @quantitative-technologies, we're in the process of removing the AdamW implementation, as you have seen it is deprecated. It will be removed in v5.
